adapter/pkg/messaging/notificationListener.go

```diff
@@ -261,69 +284,104 @@ func handlePolicyEvents(data []byte, eventType string) {
 			RateLimitTimeUnit: subscriptionPolicyEvent.RateLimitTimeUnit, StopOnQuotaReach: subscriptionPolicyEvent.StopOnQuotaReach,
 			TenantDomain: subscriptionPolicyEvent.TenantDomain, TimeStamp: subscriptionPolicyEvent.TimeStamp}
 
-		if subscriptionPolicyEvent.Event.Type == "POLICY_CREATE" {
+		if subscriptionPolicyEvent.Event.Type == policyCreate {
 			subscription.SubPolicyList.List = append(subscription.SubPolicyList.List, subscriptionPolicy)
-		} else if subscriptionPolicyEvent.Event.Type == "POLICY_UPDATE" {
+		} else if subscriptionPolicyEvent.Event.Type == policyUpdate {
 			subscription.SubPolicyList.List = removeSubPolicy(subscription.SubPolicyList.List, subscriptionPolicyEvent.PolicyID)
 			subscription.SubPolicyList.List = append(subscription.SubPolicyList.List, subscriptionPolicy)
-		} else if subscriptionPolicyEvent.Event.Type == "POLICY_DELETE" {
+		} else if subscriptionPolicyEvent.Event.Type == policyDelete {
 			subscription.SubPolicyList.List = removeSubPolicy(subscription.SubPolicyList.List, subscriptionPolicyEvent.PolicyID)
 		}
 		xds.UpdateEnforcerSubscriptionPolicies(xds.GenerateSubscriptionPolicyList(subscription.SubPolicyList))
 	}
 }
 
 func removeApplication(applications []resourceTypes.Application, id int32) []resourceTypes.Application {
-	index := 0
-	for _, i := range applications {
-		if i.ID != id {
-			applications[index] = i
-			index++
+	deleteIndex := -1
```

Comment:
`TODO: (VirajSalaka) Improve the search logic with binary search mechanism`

Reply:
Lets add the TODO in the code

Reply:
added via https://github.com/wso2/product-microgateway/pull/1651/commits/39630098027060f89dcb70ec211525197f7b1f3f